### DISCUSSION TITLE
JSON Created with ReadDWG

### DISCUSSION THREAD

**Discussion by W-Power** [LibreDWG/libredwg#654] [Q&A] [answered]
Hello,
I've converted some .dwg files to JSON using the dwgread.exe and I'm happy with the results.
The JSON file is then used to recreate (parts of) the drawing in another tool I'm building in python. The drawing files are used as reference images in the background of my python tool. I have had excellent results with the basic things I've done so far, but I could use some help understanding all the information stored in the JSON file. For the sake of prototyping, I'm simply grabbing all the 'LINE' entities I find in the CAD file and creating tkinter canvas lines using the start and end points. Here's a screengrab of what the drawing looks like in tkinter, using ONLY the 'line' elements from the attached .json file: 
![image](https://user-images.githubusercontent.com/28224573/222763208-b9eb4d27-c984-464d-beda-dbbf2f9fb42f.png)

I've poked around in the Open Design specification for DWG files, but honestly its information overload, and I'm having a hard time sorting out which data I might actually care about. 

For example, in the attached JSON file, lets take lines 537-630 as shown below.
Can anyone tell me what the values under 'handle','ownerhandle', and 'entries' mean? I really don't mind doing my own homework... I just need a push in the right direction. Is there an 'idiots guide to dwg json' type document out there?
```
    {
      "object": "LAYER_CONTROL",
      "index": 1,
      "type": 50,
      "handle": [0, 2],
      "size": 112,
      "bitsize": 42,
      "is_xdic_missing": 0,
      "ownerhandle": [4, 0, 0, 0, 0],
      "entries": [
        [2, 1, 16, 16, 0],
        [2, 0, 0, 0, 0],
        [2, 0, 0, 0, 0],
        [2, 0, 0, 0, 0],
        [2, 0, 0, 0, 0],
        [2, 0, 0, 0, 0],
        ...(truncated for clarity)
    }
```
Thanks!
[TestCAD.txt](https://github.com/LibreDWG/libredwg/files/10883770/TestCAD.txt)

**Comment by rurban** [comment, marked as answer]
Nope. The JSON just reflects the structure of the DWG and how we store all the fields.

The handles arrays changes pretty often, you have to look at the header of out_json.c to see what exactly is emitted for the handles.

E.g search for VALUE_HANDLE and see ARGS_HREF11 and ARGS_HREF.

ARGS_HREF11 has ref->handleref.size, ref->r11_idx, ref->absolute_ref
ARGS_HREF has ref->handleref.code, ref->handleref.size, ref->handleref.value, ref->absolute_ref
